packages/asset-swapper/src/utils/market_operation_utils/sampler_operations.ts

```diff
@@ -708,6 +710,62 @@ export class SamplerOperations {
         });
     }
 
+    public getUniswapV3SellQuotes(
+        router: string,
+        quoter: string,
+        tokenAddressPath: string[],
+        takerFillAmounts: BigNumber[],
+        source: ERC20BridgeSource = ERC20BridgeSource.UniswapV3,
+    ): SourceQuoteOperation<UniswapV3FillData> {
+        return new SamplerContractOperation({
+            source,
+            contract: this._samplerContract,
+            function: this._samplerContract.sampleSellsFromUniswapV3,
+            params: [quoter, tokenAddressPath, takerFillAmounts],
+            callback: (callResults: string, fillData: UniswapV3FillData): BigNumber[] => {
+                const [samples, paths] = this._samplerContract.getABIDecodedReturnData<[BigNumber[], string[]]>(
+                    'sampleSellsFromUniswapV3',
+                    callResults,
+                );
+                fillData.router = router;
+                fillData.tokenAddressPath = tokenAddressPath;
+                fillData.pathAmounts = paths.map((uniswapPath, i) => ({
+                    uniswapPath,
+                    inputAmount: takerFillAmounts[i],
```

Comment:
```
inputAmount: takerFillAmounts[i],
```

Why do we keep this? Wouldn't the first path get the lowest split `takerAmount` and the second path get the second lowest split `takerAmount`? Not sure I understand

Reply:
yeah I think I can just grab the uniswap path from the last subfill's fillData in the collapsed path.

Reply:
the top-level fillData from the collapsed path is probably a lie anyway.

Reply:
Wait, `getUniswapV3SellQuotes()` only emits a single `fillData` for all samples, so every fill shares the same `fillData`, so this hack is still necessary to find the appropriate uniswap path. We can/should refactor such that a distinct `fillData` gets returned with each sample later to make this neater.